beacon/client/http-provider.go

```diff
@@ -387,12 +414,31 @@ func getRequestReader(ctx context.Context, requestPath string, providerAddress s
 	response, err := client.Do(req)
 	if err != nil {
 		// Remove the query for readability
-		trimmedPath, _, _ := strings.Cut(path, "?")
 		return nil, 0, fmt.Errorf("error running GET request to [%s]: %w", trimmedPath, err)
 	}
 	return response.Body, response.StatusCode, nil
 }
 
+// Log a request and add HTTP tracing to the context if the logger has it enabled
+func logRequest(ctx context.Context, methodName string, path string) context.Context {
+	logger, _ := log.FromContext(ctx)
+	if logger == nil {
+		return ctx
+	}
+
+	logger.Debug("Calling BN request",
+		slog.String(log.MethodKey, methodName),
+		slog.String(log.PathKey, path),
```

Comment:
would be nice to log the host, too...

Reply:
The host is included in the path; it's the whole URL minus the query because queries can get unwieldy for certain calls.

Reply:
well i leave it up to you but i would probably parse it using net/url so that we can leverage structured logging fully